optimum/intel/ipex/inference.py

```diff
@@ -20,6 +30,37 @@
     import intel_extension_for_pytorch as ipex
 
 
+def ordered_inputs(inputs: str, model: PreTrainedModel):
+    """
+    Order input dict and convert input dict to tuple since jit traced model only support tuple input.
+    """
+    if hasattr(model, "forward"):
+        sig = inspect.signature(model.forward)
+    else:
+        sig = inspect.signature(model.call)
+
+    return tuple(inputs[key] for key in sig.parameters if inputs.get(key, None) is not None)
+
+
+def prepare_jit_inputs(model: PreTrainedModel, task: str):
+    """
+    Prepare tuple inputs for jit trace model
+    """
+    task = _TASK_ALIASES[task]

```

Comment:
The model is traced once, but shouldn't it be done twice when `use_cache` is set to `True` ? If not, what happens for the model first generation (when no `past_key_values` provided), in this case is the original model used instead ?

Reply:
Yes, the original model will be used when predicting the first token when `use_cache=True`. There is no need to save two traced models because it barely helps with latency and needs extra memory.

Reply:
Could you share the speed up you obtained for the different cases ? (if you have them). Also what are the speed up observed in comparison to generation without IPEX ? This could be intereting to have in the tests as well (to verify that we always get the expected speed up) what do you think ?

Reply:
cc @mfuntowicz